src/mlpack/core/data/image_info.hpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * @file image_info.hpp
+ * @author Mehul Kumar Nirala
+ *
+ * An image information holder.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_IMAGE_INFO_HPP
+#define MLPACK_CORE_DATA_IMAGE_INFO_HPP
+
+#ifdef HAS_STB // Compile this only if stb is present.
+
+#include <mlpack/prereqs.hpp>
+
+#include "extension.hpp"
+
+#define STB_IMAGE_STATIC
+#define STB_IMAGE_IMPLEMENTATION
+#include <stb_image.h>
+
+#define STB_IMAGE_WRITE_STATIC
+#define STB_IMAGE_WRITE_IMPLEMENTATION
+#include <stb_image_write.h>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * Checks if the given image filename is supported.
+ *
+ * @param filename Name of the image file.
+ * @return Boolean value indicating success if it is an image.
+ */
+inline bool ImageFormatSupported(const std::string& fileName,
+                                 bool save = false);
+
+class ImageInfo
+{
+ public:
+  /**
+   * Instantiate the ImageInfo object with the given image width, height,
+   * channels parameter.
+   *
+   * @param width Image width.
+   * @param height Image height.
+   * @param channels number of channels in the image.

```

Comment:
Just noticed, this is missing the ``quality`` parameter.

Reply:
okay.